src/main/java/rx/internal/schedulers/NewThreadWorker.java

```diff
@@ -112,7 +124,7 @@ static void purgeExecutors() {
      * @return true if the policy was successfully enabled 
      */
     public static boolean tryEnableCancelPolicy(ScheduledExecutorService exec) {
-        if (!PURGE_FORCE) {
+        if (!PURGE_FORCE && (!IS_ANDROID || IS_ANDROID_AND_SUPPORTS_SET_REMOVE_ON_CANCEL_POLICY)) {
```

Comment:
I'd change the entire method as well to do some caching and avoid looping at all if applied to the default `ScheduledThreadPoolExecutor` that is returned from `Executors.newScheduledThreadPool()`:

```java
private static volatile Object setOnCancelPolicyMethod;
private static final Object SET_ON_CANCEL_NOT_SUPPORTED = new Object();

public static boolean tryEnableCancelPolicy(ScheduledExecutorService exec) {
    if (!PURGE_FORCE && NOT_ANDROID_OR_SUPPORTS_SET_ON_CANCEL_POLICY) {
        final boolean executorPool = exec instanceof ScheduledThreadPoolExecutor;
        Method methodToCall;
        if (executorPool) {
            final Object localSetOnCancelPolicyMethod = setOnCancelPolicyMethod;
            if (localSetOnCancelPolicyMethod == SET_ON_CANCEL_NOT_SUPPORTED) {
                return false;
            }
            if (localSetOnCancelPolicyMethod == null) {
                Method m = findSetOnCancelPolicyMethod(exec);
                
                setOnCancelPolicyMethod = m != null ? m : SET_ON_CANCEL_NOT_SUPPORTED;
                
                methodToCall = m;
            } else {
                methodToCall = (Method)localSetOnCancelPolicyMethod;
            }
        } else {
            methodToCall = findSetOnCancelPolicyMethod(exec);
        }
        
        if (methodToCall != null) {
            try {
                methodToCall.invoke(exec, true);
                return true;
            } catch (Exception ex) {
                RxJavaPlugins.getInstance().getErrorHandler().handleError(ex);
            }
        }

    }
    return false;
}

private static Method findSetOnCancelPolicyMethod(ScheduledExecutorService exec) {
    for (final Method m : exec.getClass().getMethods()) {
        if (m.getName().equals("setRemoveOnCancelPolicy")
                && m.getParameterTypes().length == 1
                && m.getParameterTypes()[0] == Boolean.TYPE) {
            return m;
        }
    }
    return null;
}
```